packages/aws-cdk-lib/aws-s3/lib/bucket.ts

```diff
@@ -1882,6 +1886,26 @@ export class Bucket extends BucketBase {
     }
   }
 
+  /**
+   * Set the log group on the singleton AutoDeleteObjects provider lambda.
+   * Note: Calling this multiple times will override the previously set log group
+   *
+   * In the case of no buckets with `autoDeleteObjects: true` exist in the stack,
+   * this will throw an exception as no AutoDeleteObjects provider lambda will exist
+   * in the stack.
+   *
+   * @param scope the stack with bucket(s) with `autoDeleteObjects: true`.
+   * @param logGroup the log group to use on the lambda.
+   */
+  public static setAutoDeleteObjectsLogGroup(scope: Construct, logGroup: logs.ILogGroup): void {
+    const provider = AutoDeleteObjectsProvider.getProvider(scope, AUTO_DELETE_OBJECTS_RESOURCE_TYPE);
+
+    if (provider === undefined) {
```

Comment:
This feels weird to me where I would expect users to do something 
```
s3.Bucket.setAutoDeleteObjectsLogGroup(this, new logs.LogGroup(this, 'MyLogGroup', {
  logGroupName: 'MyLogGroup',
  retention: logs.RetentionDays.FIVE_YEARS
}))

const bucket = new s3.Bucket(this, 'MyTempFileBucket', {
  removalPolicy: cdk.RemovalPolicy.DESTROY,
  autoDeleteObjects: true,
});
```

Reply:
Implementation is updated such that the order of the method calls no longer matters.